### PR TITLE
Try fix begin block handler btc staking

### PR DESCRIPTION
**Pull request by KonradStaniec**

### CODE REVIEW

File: x/btcstaking/keeper/keeper.go [hunk 1]
```diff
@@ -100,19 +96,38 @@ func (k Keeper) BeginBlocker(ctx context.Context) error {
...
+	if len(fpDistMap) == 0 {
 		return nil
 	}
 
-	// get top N finality providers and set their voting power to KV store
-	k.setCurrentTopNVotingPower(ctx, fpPowerMap)
+	var distInfoSlice []*types.FinalityProviderDistInfo = make([]*types.FinalityProviderDistInfo, len(fpDistMap))
+	var i = 0
+	for _, distMap := range fpDistMap {
+		distInfoSlice[i] = distMap
```
Comment (SebastianElvis) [inline x/btcstaking/keeper/keeper.go:106]: due to the `for` semantics, will `distInfoSlice[i]` points to the same `distMap`  for every `i` eventually?

File: x/btcstaking/keeper/voting_power_table.go [hunk 2]
```diff
@@ -66,24 +66,33 @@ func (k Keeper) IterateActiveFPsAndBTCDelegations(ctx context.Context, handler f
...
+func (k Keeper) IterateActiveFPsAndBTCDelegationsAlternative(ctx context.Context, handler func(fp *types.FinalityProvider, btcDel *types.BTCDelegation) bool) {
+	btcDelIter := k.btcDelegationStore(ctx).Iterator(nil, nil)
+
+	defer btcDelIter.Close()
+	for ; btcDelIter.Valid(); btcDelIter.Next() {
+		var btcDel types.BTCDelegation
+		k.cdc.MustUnmarshal(btcDelIter.Value(), &btcDel)
+
+		for _, fpPk := range btcDel.FpBtcPkList {
+			fp, err := k.GetFinalityProvider(ctx, fpPk)
```
Comment (SebastianElvis) [inline x/btcstaking/keeper/voting_power_table.go:78]: This way each `fp` is retrieved for `m` times where `m` is the number of BTC delegations under this `fp`. So the DB access might not be optimal

File: x/btcstaking/keeper/voting_power_table.go [hunk 3]
```diff
@@ -66,24 +66,33 @@ func (k Keeper) IterateActiveFPsAndBTCDelegations(ctx context.Context, handler f
...
-	// set voting power for each active finality providers
-	for _, fp := range activeFps {
-		k.SetVotingPower(ctx, fp.BtcPk.MustMarshal(), babylonTipHeight, fp.VotingPower)
+			if !fp.IsSlashed() {
+				shouldContinue := handler(fp, &btcDel)
+
+				if !shouldContinue {
+					return
+				}
+			}
```
Comment (SebastianElvis) [inline x/btcstaking/keeper/voting_power_table.go:94]:
```suggestion
            shouldContinue := handler(fp, &btcDel)

            if !shouldContinue {
                return
            }
```

**Comment by KonradStaniec** [comment]
close in favour of - https://github.com/babylonchain/babylon/pull/453

**Pull request closed without merging**